sora/stats/core.py

```diff
@@ -0,0 +1,408 @@
+import numpy as np
+
+# try import ipython display for table display in notebooks
+try:
+    from IPython.display import HTML, display
+    from IPython import __version__
+    HAS_IPYTHON = int(__version__[0]) >= 7
+except ImportError:
+    HAS_IPYTHON = False
+
+
+def _in_ipynb():
+    '''Check if the code is running on an ipython notebook'''
+    try:
+        cfg = get_ipython().config 
```

Comment:
Yes. That's why first the function "tries" to execute the function. This private function helps to identify if the code is running inside a jupyter notebook, where it will be already in global context. The function itself only returns True or False.